### ISSUE TITLE
CardStack doesn't properly remount in 1.0.0-rc.1

### ISSUE THREAD

**Issue by Vlad-Zhukov** [LeoLeBras/react-router-navigation#24]
I've got the following setup (pseudocode):
```js
Switch
    Route1 render={() =>
        Navigation1
            Card1
            Card2
    Route2 render={() =>
        Navigation2
            Card3
            Card4
```

When I navigate to Route1 is works just fine. But when I try to navigate from Route1 to Route2 it throws with this error: 
```
Element type is invalid: expected a string (for built-in components) or 
a class/function (for composite components) but got: null. 

Check the render method of `SceneView`.
```

What happens is React __does not__ remount the `Navigation`, so [constructor from `CardStack`](https://github.com/LeoLeBras/react-router-navigation/blob/next/modules/CardStack.js#L37) doesn't run and `componentWillReceiveProps` does, but it sets an empty array to `state.cards`.

Versions of libs:
- react-router-navigation@1.0.0-rc.1
- react@16.0.0-alpha.6
- react-native@0.44.2

**Comment by Vlad-Zhukov** [comment]
As a workaround I render `Navigation`s with unique `key`s, I believe it should by default.

**Comment by LeoLeBras** [comment]
This should be fixed in `1.0.0-rc.2`
